components/dashboard/src/admin/UserDetail.tsx

```diff
@@ -290,8 +360,92 @@ export default function UserDetail(p: { user: User }) {
                         {renderUserBillingProperties()}
                     </div>
                 </div>
+
                 <WorkspaceSearch user={user} />
             </PageWithAdminSubMenu>
+            <Modal
+                visible={editSpendingLimit}
+                onClose={() => setEditSpendingLimit(false)}
+                title="Update Usage Limit"
+                onEnter={() => false}
+                buttons={[
+                    <button
+                        className="secondary"
+                        disabled={usageLimit === costCenter?.spendingLimit}
+                        onClick={async () => {
+                            if (usageLimit !== undefined) {
+                                await getGitpodService().server.adminSetUsageLimit(
+                                    AttributionId.render(AttributionId.create(user)),
+                                    usageLimit || 0,
+                                );
+                                setUsageLimit(undefined);
+                                initialize();
+                                setEditSpendingLimit(false);
+                            }
+                        }}
+                    >
+                        Save
+                    </button>,
+                ]}
+            >
+                <p>Update the spending limit for this account.</p>
+                <div className="flex flex-col">
+                    <input
+                        type="number"
+                        min={usage?.creditsUsed}
+                        max={500000}
+                        title="Update Usage Limit"
+                        value={usageLimit}
+                        onChange={(event) => setUsageLimit(Number.parseInt(event.target.value))}
+                    />
+                </div>
+            </Modal>
+            <Modal
+                onEnter={() => false}
+                visible={editAddCreditNote}
+                onClose={() => setEditAddCreditNote(false)}
+                title="Add Credits"
+                buttons={[
+                    <button
+                        className="secondary"
+                        disabled={creditNote.credits === 0 || !creditNote.note}
+                        onClick={async () => {
+                            if (creditNote.credits !== 0 && !!creditNote.note) {
+                                await getGitpodService().server.adminAddUsageCreditNote(
+                                    AttributionId.render(AttributionId.create(user)),
+                                    creditNote.credits,
+                                    creditNote.note,
+                                );
+                                setEditAddCreditNote(false);
+                                setCreditNote({ credits: 0 });
+                                initialize();
+                            }
+                        }}
+                    >
+                        Save
```

Comment:
**question**: Probably not the best line to comment about this, but when a user upgrades to UBP, the error returned when trying to change the usage limit from the admin dashboard below 1000 mentions again the 500 limit. Should this become 1000 or shall we change the error to become more generic?